lib/SYCL/inSPIRation.cpp

```diff
@@ -238,8 +271,24 @@ struct inSPIRation : public ModulePass {
     for (auto &F : M.functions()) {
       // Only consider definition of SYCL kernels
       // \todo Put SPIR calling convention on declarations too
-      if (!F.isDeclaration() && sycl::isKernel(F))
-        kernelSPIRify(F);
+      if (!F.isDeclaration()) {
+        if (sycl::isKernel(F)) {
+          kernelSPIRify(F);
+	  renameBlockname(F);
+	} else {
+          // For function is called in SYCL kernels
+          // Put SPIR calling convention
+          for (Use &U : F.uses()) {
+            CallSite CS(U.getUser());
+            if (CS.getInstruction() != nullptr) {
+              if (sycl::isKernel(*(CS.getInstruction()->getParent()->getParent()))) {
+                kernelCallFuncSPIRify(F);
+	        renameBlockname(F);
+	      }
+	    }
+          }
+        }
+      }
```

Comment:
I have the feeling that what we actually need is to do this on the transitive closure of all the functions called from a kernel.
So is it in the opposite direction?
And anyway, since after kernel code selection, there should be only functions that are called by some kernel or it would be intrinsics.
So I guess a lot of things could be simplified.

Reply:
Yes, what we want to do is to get transitive closure of functions called by kernel. And now it is doing in the opposite way: finding the call site which is from kernel.
And yes! After kernel code selection, there should be only functions called by kernels or intrinsics. It can be simplified to only test whether the function is intrinsic or not. Thanks! :-)